shepherd/Code.py

```diff
@@ -36,5 +59,123 @@ def assign_code_effect():
     Assign each code to a random effect
     '''
     for i in range(16):
-        code_effect[codes[i]] = random.choice(effect_list)
+        code_effect[codes[i]] = (EFFECTS.TWIST if random.random() <
+                                 CONSTANTS.TWIST_CHANCE else
+                                 EFFECTS.SPOILED_CANDY)
     return code_effect
+
+def rotate(numbers):
```

Comment:
Needs to be rewritten for clarity. It's hard to tell if some of these are correct.